cel/src/context.rs

```diff
@@ -76,20 +79,45 @@ impl Context<'_> {
         }
     }
 
+    pub fn set_variable_resolver(&mut self, r: Arc<dyn VariableResolver>) {
```

Comment:
Why use a setter as opposed to requiring the variable resolver to be specified when constructing a new context?

Current
```rust
let mut ctx = Context::default();
ctx.set_variable_resolver(&resolver);
```

Alternative
```rust
let ctx = Context::with_resolver(&resolver);
```

Reply:
Yes, fair point I do see the use-case where you want to set it on the child. 

I guess I'm not opposed to this, I'm just used to seeing lots and lots of method chaining in Rust.

```rust
let ctx = Context::default();
let ctx = ctx.new_inner_scope()
    .with_variable_resolver(&resolver);
```

But I'm fine with leaving as is for now since that's more of a nit